libs/dmlf/src/networkers/muddle_learner_networker.cpp

```diff
@@ -75,10 +75,32 @@ MuddleLearnerNetworker::MuddleLearnerNetworker(const std::string &cloud_config,
   mud_->SetPeerSelectionMode(muddle::PeerSelectionMode::KADEMLIA);
 
   std::unordered_set<std::string> initial_peers;
-  if (instance_number > 0)
+  auto                            config_peers = doc.root()["peers"];
+
+  auto config_peer_count = config_peers.size();
+  for (std::size_t peer_number = 0; peer_number < config_peer_count; peer_number++)
+  {
+    if (peer_number != instance_number)
+    {
+      peers_.emplace_back(config_peers[peer_number]["pub"].As<std::string>());
+    }
+  }
+
+  if (config_peer_count < 10)
   {
     initial_peers.insert(doc.root()["peers"][0]["uri"].As<std::string>());
   }
+  else
+  {
+    for (unsigned int i = 0; i < 10; i++)
+    {
+      auto peer_number = (instance_number + 1 + i) % config_peer_count;
+      if (peer_number != instance_number)
```

Comment:
If we change the loop condition to i <= 10 then we can remove this if statement

Reply:
No, we can't there's a modulo in there to cause trouble.